whitenoise/middleware.py

```diff
@@ -168,3 +189,20 @@ def get_static_url(self, name):
             return decode_if_byte_string(staticfiles_storage.url(name))
         except ValueError:
             return None
+
+
+class LazyWhiteNoiseMiddleware(WhiteNoiseMiddleware):
```

Comment:
I would personally implement this feature directly into the original middleware class, and then have it toggleable by a `settings.py` value.